### PR TITLE
remove chance of removing mutations from other trees

### LINKED ISSUE #60126: Mutation trees should not be mutually exclusive
### Is your feature request related to a problem? Please describe.

Currently, if you cross mutation trees and take mutations from different trees, the mutations from the new tree will remove mutations from the first tree. I don't think this should happen. There are already several barriers in place to stop you from doing this:
1 - mutations are costlier and harder to manufacture and apply now. It's much more expensive to get a ton of mutations.
2 - mutations already have a mechanic for this - conflicting mutations. If you get some awesome chitin armor from Insect, you might turn it into less good fur if you start taking Feline mutagen for example.
3 - mutations have an innate mechanic which makes mutating a worse prospect the more you mutate, instability. Unless you specifically have Robust Genetics and wait a very long time between each mutation (roughly a month per mutagenic primer) continually mutating will cause you to start getting more and more "bad" mutations.
4 - there is a specific malus which causes mutations to be more likely to be bad if you currently have a lot of mutagen vitamins from other trees.
5 - enemy mutants are often hybrids of multiple mutation trees. Shelled mutants are a mix of cephalopod, gastropod and possibly even more. These enemies *are* homicidally insane which definitely implies mutating like that has some serious cognitive side effects but it at least implies mixing trees is possible. This is probably the weakest of my arguments but I figured it bore mentioning

### Solution you would like.

Allow cross contamination of mutation trees without auto-pruning mutations from other trees.

### Describe alternatives you have considered.

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by anoobindisguise**
#### Summary
Balance "mutations don't remove mutations from other trees unless they would conflict"

#### Purpose of change
Fixes #60126
TLDR: There are a lot of barriers in place already to hybridizing mutation trees in place already and there doesn't need to be an additional 25% chance per mutation out of the tree of mutation line you are currently mutating to be removed automatically. 

#### Describe the solution
Remove entirely. Leave standard mutation conflicts, instability and mutation costliness as barriers to hybridizing the tree.

#### Describe alternatives you've considered
none tbh. I feel strongly about this

#### Testing
1. Spawned new character.
2. Injected with Feline Mutagen.
3. Turned partially into a cat with no issues.
4. Injected with Cephalopod Mutagen.
5. Started to turn into a squid. Didn't lose any of my Feline mutations except when Cephalopod wanted to override them (my Retractable Claws turned into regular claws as part of gaining tentacle arms, my fur turned into rough skin so it could become a shell, but my cat ears didn't ever go away because Cephalopod doesn't counter those)
6. Injected myself with Purifier and purified back to baseline. No errors or crashes.

### CODE REVIEW

**Comment by bombasticSlacks** [comment]
I think that this is reasonable. 

HOWEVER I would ask the question does this exist so that human mutagen clears your mutations. Like without this does purifier break.

**Comment by anoobindisguise** [comment]
> I think that this is reasonable.
> 
> HOWEVER I would ask the question does this exist so that human mutagen clears your mutations. Like without this does purifier break.

I believe purifier should still work. It tries to mutate you towards specific hidden mutations that remove your existing ones. And it worked before the 1/4 chance to remove cross-tree muts was added, as there was a time where that wasn't the case under the new system. However it definitely bears testing first.

**Comment by actually-a-cat** [comment]
```cpp
        } else {
            // Remove existing mutations that don't fit into our category
            if( !downgrades.empty() ) {
                size_t roll = rng( 0, downgrades.size() + 4 );
                if( roll < downgrades.size() ) {
                    remove_mutation( downgrades[roll] );
                    return;
                }
            }
```
This bit also needs to go.

**Comment by anoobindisguise** [comment]
> ```c++
>         } else {
>             // Remove existing mutations that don't fit into our category
>             if( !downgrades.empty() ) {
>                 size_t roll = rng( 0, downgrades.size() + 4 );
>                 if( roll < downgrades.size() ) {
>                     remove_mutation( downgrades[roll] );
>                     return;
>                 }
>             }
> ```
> 
> This bit also needs to go.

good catch, thank you.

**Comment by anoobindisguise** [comment]
earlier today I did test this in game and left my notes in the testing section of the PR, it's functional and I didn't find any errors or weirdness.
